retweets.go

```diff
@@ -51,3 +52,89 @@ func retweetsLookup(ctx context.Context, c *client, id string, opt ...*RetweetsL
 
 	return &retweetsLookup, nil
 }
+
+type Body struct {
+	TweetID string `json:"tweet_id"`
+}
+
+func postRetweet(ctx context.Context, c *client, uid string, tid string) (*PostRetweetResponse, error) {
+	if uid == "" {
+		return nil, errors.New("post retweet by uid: uid parameter is required")
+	}
+	postRetweetPath := timeline + "/" + uid + "/retweets"
+
+	if tid == "" {
+		return nil, errors.New("post retweet by tid: tid parameter is required")
+	}
+	body := &Body{
+		TweetID: tid,
+	}
+	jsonStr, err := json.Marshal(body)
+	if err != nil {
+		return nil, errors.New("post retweet: can not marshal")
+	}
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodPost, postRetweetPath, bytes.NewBuffer(jsonStr))
+	if err != nil {
+		return nil, fmt.Errorf("post retweet new request with ctx: %w", err)
+	}
+	req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", c.bearerToken))
+	req.Header.Set("Content-Type", "application/json")
+
+	resp, err := c.client.Do(req)
+	if err != nil {
+		return nil, fmt.Errorf("post retweet response: %w", err)
+	}
+	defer resp.Body.Close()
+
+	var postRetweet PostRetweetResponse
+	if err := json.NewDecoder(resp.Body).Decode(&postRetweet); err != nil {
+		return nil, fmt.Errorf("post retweet decode: %w", err)
+	}
+	if resp.StatusCode != http.StatusOK {
+		return &postRetweet, &HTTPError{
+			APIName: "post retweet",
+			Status:  resp.Status,
+			URL:     req.URL.String(),
+		}
+	}
+
+	return &postRetweet, nil
+}
+
+// stid = "source_tweet_id"
+func deleteRetweet(ctx context.Context, c *client, id string, stid string) (*DeleteRetweetResponse, error) {
+	if id == "" {
+		return nil, errors.New("delete retweet by id: id parameter is required")
+	}
+	if stid == "" {
+		return nil, errors.New("delete retweet by source_tweet_id: source_tweet_id parameter is required")
+	}
+	deleteRetweetPath := timeline + "/" + id + "/retweets/" + stid
```

Comment:
Please don't reuse deferent API endpoint.
This API is different from the timeline API.